File: DevExpress-Demo/VB/LayoutControlDemo.Wpf/Data/AdvancedGroupedLayoutData_FluentAPI.vb
Imports DevExpress.Mvvm.DataAnnotations
Imports System.ComponentModel.DataAnnotations
Imports System.Windows.Controls

Namespace LayoutControlDemo

    <MetadataType(GetType(AdvancedGroupedLayoutMetadata))>
    Public Class AdvancedGroupedLayoutData_FluentAPI

        Public Property AddressLine1 As String

        Public Property AddressLine2 As String

        Public Property BirthDate As Date

        Public Property Email As String

        Public Property FirstName As String

        Public Property Gender As Gender

        Public Property Group As String

        Public Property HireDate As Date

        Public Property LastName As String

        Public Property Phone As String

        Public Property Salary As Decimal

        Public Property Title As String
    End Class

    Public Module AdvancedGroupedLayoutMetadata

        Public Sub BuildMetadata(ByVal builder As MetadataBuilder(Of AdvancedGroupedLayoutData_FluentAPI))
            builder.DataFormLayout().Group("Name", Orientation.Horizontal).ContainsProperty(Function(x) x.FirstName).ContainsProperty(Function(x) x.LastName).EndGroup().TabbedGroup("Tabs").Group("Job").ContainsProperty(Function(x) x.Group).ContainsProperty(Function(x) x.Title).ContainsProperty(Function(x) x.HireDate).ContainsProperty(Function(x) x.Salary).EndGroup().Group("Contact").ContainsProperty(Function(x) x.Phone).ContainsProperty(Function(x) x.Email).GroupBox("Address").ContainsProperty(Function(x) x.AddressLine1).ContainsProperty(Function(x) x.AddressLine2).EndGroup().EndGroup().EndGroup().GroupBox("Personal", Orientation.Horizontal).ContainsProperty(Function(x) x.Gender).ContainsProperty(Function(x) x.BirthDate).EndGroup()
            builder.Property(Function(x) x.AddressLine1).DisplayShortName(String.Empty)
            builder.Property(Function(x) x.AddressLine2).DisplayShortName(String.Empty)
            builder.Property(Function(x) x.BirthDate).DisplayName("Birth date")
            builder.Property(Function(x) x.FirstName).DisplayName("First name")
            builder.Property(Function(x) x.LastName).DisplayName("Last name")
            builder.Property(Function(x) x.HireDate).DisplayName("Hire date")
            builder.[Property](Function(x) x.Phone).PhoneNumberDataType()
            builder.[Property](Function(x) x.Salary).CurrencyDataType()
        End Sub
    End Module
End Namespace

File: DevExpress-Demo/CS/LayoutControlDemo.Wpf/Data/AdvancedGroupedLayoutData_FluentAPI.cs
using DevExpress.Mvvm.DataAnnotations;
using System;
using System.ComponentModel.DataAnnotations;
using System.Windows.Controls;

namespace LayoutControlDemo {
    [MetadataType(typeof(AdvancedGroupedLayoutMetadata))]
    public class AdvancedGroupedLayoutData_FluentAPI {
        public string AddressLine1 { get; set; }
        public string AddressLine2 { get; set; }
        public DateTime BirthDate { get; set; }
        public string Email { get; set; }
        public string FirstName { get; set; }
        public Gender Gender { get; set; }
        public string Group { get; set; }
        public DateTime HireDate { get; set; }
        public string LastName { get; set; }
        public string Phone { get; set; }
        public decimal Salary { get; set; }
        public string Title { get; set; }
    }
    public static class AdvancedGroupedLayoutMetadata {
        public static void BuildMetadata(MetadataBuilder<AdvancedGroupedLayoutData_FluentAPI> builder) {
            builder.DataFormLayout()
               .Group("Name", Orientation.Horizontal)
                   .ContainsProperty(x => x.FirstName)
                   .ContainsProperty(x => x.LastName)
               .EndGroup()
               .TabbedGroup("Tabs")
                   .Group("Job")
                       .ContainsProperty(x => x.Group)
                       .ContainsProperty(x => x.Title)
                       .ContainsProperty(x => x.HireDate)
                       .ContainsProperty(x => x.Salary)
                   .EndGroup()
                   .Group("Contact")
                       .ContainsProperty(x => x.Phone)
                       .ContainsProperty(x => x.Email)
                       .GroupBox("Address")
                           .ContainsProperty(x => x.AddressLine1)
                           .ContainsProperty(x => x.AddressLine2)
                       .EndGroup()
                   .EndGroup()
               .EndGroup()
               .GroupBox("Personal", Orientation.Horizontal)
                   .ContainsProperty(x => x.Gender)
                   .ContainsProperty(x => x.BirthDate)
               .EndGroup();


            builder.Property(x => x.AddressLine1)
                .DisplayShortName(string.Empty);
            builder.Property(x => x.AddressLine2)
                .DisplayShortName(string.Empty);
            builder.Property(x => x.BirthDate)
                .DisplayName("Birth date");
            builder.Property(x => x.FirstName)
                .DisplayName("First name");
            builder.Property(x => x.LastName)
                .DisplayName("Last name");
            builder.Property(x => x.HireDate)
                .DisplayName("Hire date");
            builder.Property(x => x.Phone)
                .PhoneNumberDataType();
            builder.Property(x => x.Salary)
                .CurrencyDataType();
        }
    }
}
